WordPress/src/test/java/org/wordpress/android/viewmodel/giphy/provider/TenorProviderTest.kt

```diff
@@ -29,17 +29,22 @@ import org.wordpress.android.viewmodel.giphy.provider.GifProvider.GifRequestFail
 import org.wordpress.android.viewmodel.giphy.provider.TenorProviderTestFixtures.expectedGifMediaViewModelCollection
 import org.wordpress.android.viewmodel.giphy.provider.TenorProviderTestFixtures.mockedTenorResult
 import retrofit2.Call
+import retrofit2.Callback
+import retrofit2.Response
 
+@ExperimentalCoroutinesApi
 @Config(application = TestApplication::class)
 @RunWith(RobolectricTestRunner::class)
```

Comment:
I was looking into the test and I think it could be doable without Roboelectric. I'm generally against using Roboelectric because it usually means you're not testing something right (you're testing the Roboelectric framework, not your code). Sometimes you cannot avoid it but since all you're using it for here is building the Api client (which is an interface), I think you can replace it with `MockitoJUnitRunner` and pass in mocked ApiClient (and verify you call its methods). What do you think about this approach?

Reply:
I've tried some further approaches trying to remove the Robolectric from the `TenorProviderTest`class. The Context mocking was easy going, just needed to adjust the `ApiClient.getServices(context)` inside the `TenorProvider` and mock the `getString` method. But as we talked before, I didn't find a good solution for the `Uri.parse`, my implementation options were all some kind of workaround to avoid direct `Uri` usage or way too overkill for a proper solution.

I think we'll have to choose the tradeoffs. What do you think?

Reply:
As discussed on Slack, let's keep Roboelectric in place. Thanks for the investigation!